R/fetch-coaches-votes.R

```diff
@@ -0,0 +1,134 @@
+#' Fetch Coaches Votes
+#'
+#' @description
+#' `fetch_coaches_votes` returns all coaches votes for input season/s, round/s, and/or team's matches.
+#' The function calls a core `scrape_coaches_votes` function which scrapes the AFLCA website for coaches votes
+#' for a particular season, round and competition.
+#'
+#' @param comp One of "AFLM" (default) or "AFLW"
+#' @param season Season in YYYY format. This can be an array of seasons.
+#' @param round Optional - Round number. For finals this is the number of H&A rounds plus the Finals week.
+#' @param team Optional - Team or teams whose matches should be retrieved (defaults to all teams)
+#'
+#' @return
+#' A data frame with columns: Season, Round, Finals, Home.Team, Away.Team, Player.Name, Coaches.Votes
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # Return all coaches votes across all seasons
+#' fetch_coaches_votes(comp = "AFLM", season = 2007:2021)
+#' fetch_coaches_votes(comp = "AFLW", season = 2018:2021)
+#'
+#' # Return all coaches votes for a particular round
+#' fetch_coaches_votes(comp = "AFLM", season = 2021, round = 24)
+#' fetch_coaches_votes(comp = "AFLW", season = 2021, round = 9)
+#'
+#' # Return all coaches votes for a particular team
+#' fetch_coaches_votes(comp = "AFLM", season = 2021, team = "Western Bulldogs")
+#' fetch_coaches_votes(comp = "AFLW", season = 2021, team = "Western Bulldogs)
+#'
+#' # Return all coaches votes for a particular match
+#' fetch_coaches_votes(comp = "AFLM", season = 2021, round = 24, team = "Western Bulldogs")
+#' fetch_coaches_votes(comp = "AFLW", season = 2021, round = 9, team = "Western Bulldogs)
+#' }
+#'
+#' @importFrom magrittr %>%
+#' @importFrom rlang .data
+fetch_coaches_votes <- function(comp = "AFLM", season, round = 1:27,
```

Comment:
For consistency, if you could follow the `fetch_` functions arguments as closely as possible that would be great. You aren't far away but see `fetch_lineup` as a relatively good example of this one. 

In general, each `fetch_` function has the following arguments and defaults to NULL.

- `season`
- `round_number`
- `comp`

See the example below
```
fetch_lineup <- function(season = NULL,
                          round_number = NULL,
                          comp = "AFLM",
                          source = "AFL",
                          ...) {
```